puppet/lib/puppet/util/puppetdb.rb

```diff
@@ -64,6 +66,26 @@ def submit_command(certname, payload, command_name, version)
     end
   end
 
+  # Query PuppetDB.
+  #
+  # @param certname [String] The certname this command operates on
+  # @param payload [String] payload
+  # @param command_name [String] name of command
+  # @param version [Number] version number of command
+  # @return [Hash <String, String>]
+  def query_puppetdb(query)
+    profile("Submitted query '#{query}'",
+            [:puppetdb, :query, query]) do
+      headers = { "Accept" => "application/json",
+                  "Content-Type" => "application/json; charset=UTF-8" }
+      response = Puppet::Util::Puppetdb::Http.action("/pdb/query/v4?query=#{CGI.escape(query)}",
+                                                     :query) do |http_instance, path|
+        http_instance.get(path, headers)
```

Comment:
Should we use the POST query api here, just as insurance against monster queries?